src/Conversion/ONNXToTOSA/Tensor/Flatten.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Flatten.cpp - Flatten Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX ReshapeOp to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/Interfaces/InferTypeOpInterface.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "llvm/ADT/SmallVector.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXFlattenLoweringToTOSA : public OpConversionPattern<ONNXFlattenOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXFlattenOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXFlattenOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+
+    Value input = adaptor.input();
+    int64_t axis = adaptor.axis();
+    auto inputType = input.getType().cast<ShapedType>();
+
+    auto resultType = op.getResult().getType();
+
+    if (axis < 0) {
+      axis += inputType.getRank();
```

Comment:
Is there any legalization check already that confirms that `axis` is between -rank and rank? 
Maybe you can add a comment about the behaviour of the negative axis, too, because that's a pretty surprising one.

Reply:
Yes, the onnx verifier does not allow another value. I will add a comment for that